test/unit/spriteManipulation.js

```diff
@@ -393,3 +393,46 @@ test('setPropRandom set sprite y properties between 50 and 350', async t => {
 
   nativeAPI.reset();
 });
+
+test('startMapping/stopMapping adds and removes behaviors', async t => {
+  const nativeAPI = await helpers.createDanceAPI();
+  nativeAPI.play({
+    bpm: 120,
+  });
+  nativeAPI.setAnimationSpriteSheet("CAT", 0, {}, () => {});
+
+  const sprite = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+
+  // This is 1 only because we have a behavior that we add to every sprite
+  // (which should arguably be a different concept)
+  t.equal(sprite.behaviors.length, 1);
+
+  nativeAPI.startMapping(sprite, 'x', 'bass');
+
+  t.equal(sprite.behaviors.length, 2);
+
+  // adding the same mapping again gets ignored
+  // TODO: this fails
+  // nativeAPI.startMapping(sprite, 'x', 'bass');
+  // t.equal(sprite.behaviors.length, 2);
```

Comment:
I don't know much about behaviors, is this something we need to address before merge?

Reply:
bah, that should be uncommented (and will hopefully pass)